components/raftstore-v2/src/operation/command/admin/compact_log.rs

```diff
@@ -97,7 +97,7 @@ impl CompactLogContext {
 
 impl<'a, EK: KvEngine, ER: RaftEngine, T: Transport> PeerFsmDelegate<'a, EK, ER, T> {
     pub fn on_compact_log_tick(&mut self, force: bool) {
-        if !self.fsm.peer().is_leader() {
+        if !self.fsm.peer().is_leader() || !self.fsm.peer().serving() {
```

Comment:
Is there any issue with the tick if peer is not serving?

Should we add the serving check to other ticks?

Reply:
Raft log is cleared and cannot get term.